app/routes.py

```diff
@@ -1,2 +1,242 @@
-from flask import Blueprint
+from flask import request, jsonify, abort, Blueprint, g
+from app import db
+from sqlalchemy import exc
+from app.models.task import Task
+from app.models.goal import Goal
 
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@goal_bp.before_request
+@task_bp.before_request
+def get_model_and_label():
+    """This is a function that runs before the request is processed.
+    It will set g.model and g.label based on the blueprint being accessed,
+    both of which are accessible globally to use in other functions.
+    """
+    bps = {
+        "tasks": (Task, "task"),
+        "goals": (Goal, "goal")
+    }
+    g.model, g.label = bps[request.blueprint]
+
+
+@goal_bp.errorhandler(400)
+@task_bp.errorhandler(400)
+def invalid_data(error):
+    """This is a function to handle any 400 status code errors"""
+    return jsonify({"details": "Invalid data"}), 400
+
+
+@goal_bp.route("", methods=["GET"])
+@task_bp.route("", methods=["GET"])
+def read_items():
+    """This is a route to get all saved tasks or goals
+    Optional query parameter:
+        - sort: can be "asc" or "desc" to sort tasks or goals by title
+    Returns:
+        - JSON array of tasks or goals represented as objects, optionally sorted by title
+        - 200 status code
+    """
+    model = g.model
+    sort_query = request.args.get("sort")
+    if sort_query == 'asc':
+        items = model.query.order_by(model.title).all()
+    elif sort_query == 'desc':
+        items = model.query.order_by(model.title.desc()).all()
+    else:
+        items = model.query.all()
+    items = [item.to_dict() for item in items]
+    return jsonify(items), 200
+
+
+@goal_bp.route("", methods=["POST"])
+@task_bp.route("", methods=["POST"])
+def create_item():
+    """This is a route to create a new task or goal
+    Required request body:
+        - For task creation:
+            - JSON object with title (string), description (string), and completed_at (datetime or null) keys
+        - For goal creation:
+            - JSON object with title (string)
+    Returns:
+        - If valid data provided:
+            - JSON object with task or goal data saved to the db
+            - 201 status code
+        - If invalid data provided:
+            - JSON error message
+            - 400 status code
+    """
+    model, label = g.model, g.label
+    req = request.get_json()
+
```

Comment:
Nice work handling these different types of errors to return appropriate 400 level responses